src/renderer/chunk_meshing.zig

```diff
@@ -1184,56 +1182,14 @@ pub const ChunkMesh = struct { // MARK: ChunkMesh
 
 		if(oldBlock == newBlock) {
 			self.mutex.unlock();
-			return;
+			return .noChange;
 		}
 		self.mutex.unlock();
 
 		if(oldBlock.entityDataClass()) |class| {
 			class.onBreakClient(.{_x, _y, _z}, self.chunk);
 		}
 
-		var neighborBlocks: [6]Block = undefined;
-		@memset(&neighborBlocks, .{.typ = 0, .data = 0});
-		for(chunk.Neighbor.iterable) |neighbor| {
-			const nx = x + neighbor.relX();
-			const ny = y + neighbor.relY();
-			const nz = z + neighbor.relZ();
-			if(nx & chunk.chunkMask != nx or ny & chunk.chunkMask != ny or nz & chunk.chunkMask != nz) {
-				const neighborChunkMesh = mesh_storage.getNeighborAndIncreaseRefCount(self.pos, self.pos.voxelSize, neighbor) orelse continue;
-				defer neighborChunkMesh.decreaseRefCount();
-				const index = chunk.getIndex(nx & chunk.chunkMask, ny & chunk.chunkMask, nz & chunk.chunkMask);
-				neighborChunkMesh.mutex.lock();
-				var neighborBlock = neighborChunkMesh.chunk.data.getValue(index);
-				if(neighborBlock.mode().dependsOnNeighbors) {
-					if(neighborBlock.mode().updateData(&neighborBlock, neighbor.reverse(), newBlock)) {
-						neighborChunkMesh.chunk.data.setValue(index, neighborBlock);
-						neighborChunkMesh.mutex.unlock();
-						neighborChunkMesh.updateBlockLight(@intCast(nx & chunk.chunkMask), @intCast(ny & chunk.chunkMask), @intCast(nz & chunk.chunkMask), neighborBlock, &lightRefreshList);
-						neighborChunkMesh.generateMesh(&lightRefreshList);
-						neighborChunkMesh.mutex.lock();
-					}
-				}
-				neighborChunkMesh.mutex.unlock();
-				neighborBlocks[neighbor.toInt()] = neighborBlock;
-			} else {
-				const index = chunk.getIndex(nx, ny, nz);
-				self.mutex.lock();
-				var neighborBlock = self.chunk.data.getValue(index);
-				if(neighborBlock.mode().dependsOnNeighbors) {
-					if(neighborBlock.mode().updateData(&neighborBlock, neighbor.reverse(), newBlock)) {
-						self.chunk.data.setValue(index, neighborBlock);
-						self.updateBlockLight(@intCast(nx & chunk.chunkMask), @intCast(ny & chunk.chunkMask), @intCast(nz & chunk.chunkMask), neighborBlock, &lightRefreshList);
-					}
-				}
-				self.mutex.unlock();
-				neighborBlocks[neighbor.toInt()] = neighborBlock;
-			}
-		}
-		if(newBlock.mode().dependsOnNeighbors) {
-			for(chunk.Neighbor.iterable) |neighbor| {
-				_ = newBlock.mode().updateData(&newBlock, neighbor, neighborBlocks[neighbor.toInt()]);
-			}
-		}
```

Comment:
Removing the neighbor block checking on the client side means that it has to do a round-trip on the network to e.g. connect fences.
I don't think this is a good experience, especially considering that round trip time can be 500 ms or more.

Reply:
Alright, the delay is annoying even locally, that's unfortunate. I guess getting around #1217 will require sending some flag / using separate protocol after all.

Reply:
Okay, I restored neighbor updates, unfortunately in the process I came across a memory leak and touched few more likes of code than I should have while looking for it. I promise I didn't break anything 🤞

Basically I have extracted `addSelfToLightRefreshList` and merged `neighborBlock.mode().dependsOnNeighbors` and `neighborBlock.mode().updateData(&neighborBlock, neighbor.reverse(), newBlock)` conditions into single if statement to have single else with decref